server/session/security_cookie.ts

```diff
@@ -41,8 +47,23 @@ export function getSecurityCookieOptions(
     encryptionKey: config.cookie.password,
     validate: (sessionStorage: SecuritySessionCookie | SecuritySessionCookie[]) => {
       sessionStorage = sessionStorage as SecuritySessionCookie;
+      if (sessionStorage === undefined) {
+        return { isValid: false, path: '/' };
+      }
+
+      // TODO: with setting redirect attributes to support OIDC and SAML,
+      //       we need to do additonal cookie validatin in AuthenticationHandlers.
+      // if SAML fields present
+      if (sessionStorage.saml && sessionStorage.saml.requestId && sessionStorage.saml.nextUrl) {
+        return { isValid: true, path: '/'};
+      }
+
+      // if OIDC fields present
+      if (sessionStorage.oidc ) {
```

Comment:
Out of curiosity, is OIDC related to SAML? If not, please add description for OIDC to commit message and CR description.

Reply:
ah, no, OIDC has nothing to do with SAML, I can move this to another PR as well